operator/src/test/java/oracle/kubernetes/operator/MainTest.java

```diff
@@ -833,10 +836,83 @@ public void withNamespaceList_onReadExistingNamespaces_whenDomainNamespaceRemove
     HelmAccessStub.defineVariable(HelmAccess.OPERATOR_DOMAIN_NAMESPACES, namespaceString);
     createNamespaces(4);
 
-    testSupport.runSteps(createDomainRecheck().readExistingNamespaces());
+    runCreateReadNamespacesStep();
     assertThat(logRecords, containsInfo(MessageKeys.END_MANAGING_NAMESPACE, "NS3"));
   }
 
+  @Test
+  public void withNamespaceList_changeToDedicated_onCreateReadNamespaces_nsWatchStoppedEventCreated() {
+    defineSelectionStrategy(SelectionStrategy.List);
+    String namespaceString = "NS1";
+    HelmAccessStub.defineVariable(HelmAccess.OPERATOR_DOMAIN_NAMESPACES, namespaceString);
+    createNamespaces(4);
```

Comment:
I have found that organizing tests can help readability:

1. general setup (what the world looks like before the operation)
2. the action to test
3. the assertion.


You can do that with judicious use of spaces. Here, lines 845-850 establish your starting point, and 852-854 are the action, so I would remove the blank lines in between them, and reserve those as separators only between the three sections.

If those are complex, you might consider extract them to their own methods, especially if you do the same setup in multiple tests.